From 542c19750cf9af6a64e75e66f3746bfbc9e30e92 Mon Sep 17 00:00:00 2001
From: Daniel Kondor <kondor.dani@gmail.com>
Date: Sun, 23 Nov 2025 19:37:26 +0100
Subject: [PATCH 3/4] build: add ext-foreign-toplevel-state to build

---
 src/implementations/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/implementations/CMakeLists.txt b/src/implementations/CMakeLists.txt
index b44f00e7..45de2e9a 100644
--- a/src/implementations/CMakeLists.txt
+++ b/src/implementations/CMakeLists.txt
@@ -37,6 +37,7 @@ if(WaylandScanner_FOUND)
 		cairo-dock-plasma-window-manager.c   cairo-dock-plasma-window-manager.h
 		cairo-dock-plasma-virtual-desktop.c  cairo-dock-plasma-virtual-desktop.h
 		cairo-dock-ext-workspaces.c          cairo-dock-ext-workspaces.h
+		cairo-dock-ext-toplevel.c            cairo-dock-ext-toplevel.h
 	)
 
 	ecm_add_wayland_client_protocol(
@@ -75,6 +76,10 @@ if(WaylandScanner_FOUND)
 		impl_SRCS
 		PROTOCOL "proto/ext-foreign-toplevel-list-v1.xml"
 		BASENAME "ext-toplevel-list")
+	ecm_add_wayland_client_protocol(
+		impl_SRCS
+		PROTOCOL "proto/ext-foreign-toplevel-state-v1.xml"
+		BASENAME "ext-toplevel-state")
 	ecm_add_wayland_client_protocol(
 		impl_SRCS
 		PROTOCOL "proto/plasma-virtual-desktop.xml"
-- 
2.43.0

